### PR TITLE
logging for tls pre-master key

### PR DESCRIPTION
**Pull request by hellt**
Hi @karimra 

I would like gnmic to be able to log the [TLS pre master key file](https://gitlab.com/wireshark/wireshark/-/wikis/TLS#using-the-pre-master-secret) in order to be able to dissect secured gNMI communications in wireshark

Luckily, this is easily done within go stdlib by setting `KeyLogWriter` field of a TLSConfig struct.

The main workflow for this feature I envision is as follows:

1. A user wants to see what happens on the wire when gnmic is interacts with a secured gNMI server.
2. A user then uses the gnmic as a CLI client and configures a single target by using CLI flags
3. If a user specified a flag such as `--log-tls-premaster-key` then gnmic will create a log file named as `<target-name>.tlskey.log` in the PWD
4. the user then takes this file and configures wireshark to use it in order to dissect TLS packets.

If you agree with the approach, I would like to add the flag, but I forgot which pieces I should touch to do that...

### CODE REVIEW

**Comment by karimra** [comment]
This sounds good, except that the flag name is a bit too long :)

I would put the logic of creating the writer and setting it in the TLSConfig struct in this [function](https://github.com/karimra/gnmic/blob/main/types/target.go#L48) .
Then, add a bool field (`LogTLSPreMasterKey *bool ?`) under types.TargetConfig that can be set per target or be inherited from the global flag.

To add a global flag:
- Add a variable to the [GlobalFlags struct](https://github.com/karimra/gnmic/blob/main/config/config.go#L60)
- Bind the variable to the flag name in rootCmd [here](https://github.com/karimra/gnmic/blob/main/app/app.go#L127)

**Comment by hellt** [comment]
@karimra I have added the skeleton of the flag infra, but I am not sure what needs to be done to make that global flag propagate to the target config?

**Comment by karimra** [comment]
It's done here: https://github.com/karimra/gnmic/blob/main/config/targets.go#L159,
If a value is not set under the yaml target config, apply the value from the global flag or the default one.

File: docs/user_guide/configuration_env.md [hunk 1]
```diff
@@ -10,33 +10,34 @@ For e.g to set the gNMI username, the env variable `GNMIC_USERNAME` should be se
...
+| --retry              | GNMIC_RETRY              |
+| --skip-verify        | GNMIC_SKIP_VERIFY        |
+| --timeout            | GNMIC_TIMEOUT            |
+| --tls-ca             | GNMIC_TLS_CA             |
+| --tls-cert           | GNMIC_TLS_CERT           |
+| --tls-key            | GNMIC_TLS_KEY            |
+| --tls-max-version    | GNMIC_TLS_MAX_VERSION    |
+| --tls-min-version    | GNMIC_TLS_MIN_VERSION    |
+| --tls-version        | GNMIC_TLS_VERSION        |
+| --log-tls-secret     | GNMIC_LOG_TLS_SECRET     |
```
Comment (hellt) [inline docs/user_guide/configuration_env.md:34]: it seems env vars are automatically parsed, so added this new flag here as well

**Comment by hellt** [comment]
@karimra I think it is ready for a review.
Not sure about any additional tests for that one...

File: config/targets_test.go [hunk 2]
```diff
@@ -25,16 +25,17 @@ address: 10.1.1.1
...
-				SkipVerify: &falseBool,
-				Gzip:       &falseBool,
+				Address:      "10.1.1.1:57400",
+				Name:         "10.1.1.1",
+				Password:     &adminStr,
+				Username:     &adminStr,
+				Token:        &emptyStr,
+				TLSCert:      &emptyStr,
+				TLSKey:       &emptyStr,
+				LogTLSSecret: &falseBool,
```
Comment (hellt) [inline config/targets_test.go:35]: added this line to all the below maps

**Comment by karimra** [comment]
lgtm, thanks

**Pull request merged**